acceptance: Fix event flakiness

tbg opened the pull request:
The goroutine streaming from the events API sometimes spontanously gets a
string-wrapped EOF error back right at the beginning.

We now simply retry on all errors until the context gets canceled.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/5041)

<!-- Reviewable:end -->

petermattis commented:
LGTM

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[acceptance/cluster/localcluster.go, line 532 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5041#-KCT5UI54enpIlVEBFKK:-KCT5UI54enpIlVEBFKL:-1664343443) ([raw file](https://github.com/cockroachdb/cockroach/blob/3799e97f47dc0c88280cf6c754518842d70c2eda/acceptance/cluster/localcluster.go#L532)):*
The looping in here is a bit complex. I think it would be better to move the decoding loop into a separate `monitorLoop()` which returns true if looping should be restarted. Then `monitor()` would look like:

```
if log.V(1) {
  ...
}
for monitorLoop() {
}
```

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5041#-:-KCT62bGTFSuPLIaQQ4l:676834233)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/cluster/localcluster.go, line 532 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5041#-KCT5UI54enpIlVEBFKK:-KCTISRUondIo59SdtV2:-396758155) ([raw file](https://github.com/cockroachdb/cockroach/blob/3799e97f47dc0c88280cf6c754518842d70c2eda/acceptance/cluster/localcluster.go#L532)):*
+1. current code is a bit clunky (and would be simpler if the function was named)

---

*[acceptance/cluster/localcluster.go, line 181 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5041#-KCTIDfsMA_xFn4Xp9Wl:-KCTIDfsMA_xFn4Xp9Wm:1327019541) ([raw file](https://github.com/cockroachdb/cockroach/blob/62f3df3ad3d3869dc0642e734af402dccc7472e0/acceptance/cluster/localcluster.go#L181)):*
nit: use the word channel like you do below

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5041)*
<!-- Sent from Reviewable.io -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5041#-:-KCTIfQyG_ZXMjWnD1MZ:1417088721)*
<!-- Sent from Reviewable.io -->

tbg committed:
acceptance: Fix event flakiness

The goroutine streaming from the events API sometimes spontanously gets a
string-wrapped EOF error back right at the beginning.

We now simply retry on all errors until the context gets canceled.

tschottdorf commented:
Hmm - something broke with the rebase. Probably the gRPC port change. Have to check tomorrow.

tamird commented:
Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5041)*
<!-- Sent from Reviewable.io -->
